### DISCUSSION TITLE
Notifications/homeworks

### DISCUSSION THREAD

**Discussion by mgrfilipmarek** [rine77/homeassistantedupage#51] [Show and tell]
How do you present notifications at dashboard? What kind of card is recommended to use? (#9)

**Comment by rine77** [comment]
The idea behind the notification sensor was to create a feed of notifications.
While implementing it became obious that you can use this sensor for lots of things.
For testing purposes it came into to the release with the ONLY events "EventType.HOMEWORK".

So, you can play around a bit and try some Home Assistant cards or automations and report back!

**Reply by mgrfilipmarek** [reply]
What about adding some incremental attribute for events?
I can see there for example `event_11_id: 959768` but attribute name and also its value are not consecutive (in my case).

Or is there some other way how to easily iterate our events-attributes in HA?

**Reply by mgrfilipmarek** [reply]
@rine77 any hints here, please?

**Reply by rine77** [reply]
I dont know if I get you right, but in your school event_id and number are not unique so you want to implement a integration-internal system for events, right?

**Reply by mgrfilipmarek** [reply]
You can check my screen, they are unique but not consecutive. So its not easy to read them with `for loop`. So Im missing attribute with incremental index.
So my question is: Is there easy way to iterate all the data in HA? Or we have to add incremental index.

![image](https://github.com/user-attachments/assets/c394159a-e9e1-4183-83ab-1cb9476403c0)

**Reply by rine77** [reply]
yeah this is because of the EventType-filter Event.HOMEWORK. IIRC the event ID is incrementing, but not in HA because not all events are of this HW-type.

This thing should be improved anyways (use EventType class of edupage API would be nice)

EDIT: I tend to remove the notification sensor by now because it should be better described and THEN implemented.

**Reply by mgrfilipmarek** [reply]
I would leave it there for now (better something than nothing :) ). And I will try to figure something out or maybe add helper index. I was thinking also about homeworks-calendar, but it is probably more flexible as it is now. 
Maybe @radkokeves has some idea or a scenario how he wants to use this.

**Reply by mgrfilipmarek** [reply]
Or what about to-do entity: https://developers.home-assistant.io/docs/core/entity/todo
